File: SignalR.Samples/Hubs/ConnectDisconnect/Status.js
$(function () {
    var status = $.connection.status;

    status.joined = function (id, when) {
        if ($.connection.hub.clientId === id) {
            addMessage(id, 'blue');    
        }
        
        addMessage(id + ' joined at ' + when, 'green');
    };

    status.leave = function (id, when) {
        addMessage(id + ' left at ' + when, 'red');
    };

    function addMessage(value, color) {
        $('#messages').append('<li style="background-color:' + color + ';color:white">' + value + '</li>');
    }

    $.connection.hub.start(function () {
        status.join();
    });

});

File: SignalR.Samples/Hubs/ConnectDisconnect/Status.cs
using System;
using SignalR.Hubs;

namespace SignalR.Samples.Hubs.ConnectDisconnect
{
    public class Status : Hub, IDisconnect
    {
        public void Join()
        {
            Clients.joined(Context.ClientId, DateTime.Now.ToString());
        }

        public void Disconnect()
        {
            Clients.leave(Context.ClientId, DateTime.Now.ToString());
        }
    }
}